services/src/main/scala/cromwell/pipeline/service/RunService.scala

```diff
@@ -1,43 +1,45 @@
 package cromwell.pipeline.service
 
-import java.time.Instant
-
 import cromwell.pipeline.datastorage.dao.repository.RunRepository
 import cromwell.pipeline.datastorage.dto._
 import cromwell.pipeline.model.wrapper.{ RunId, UserId }
 
+import java.time.Instant
 import scala.concurrent.{ ExecutionContext, Future }
 
 trait RunService {
 
-  def addRun(runCreateRequest: RunCreateRequest): Future[RunId]
+  def addRun(runCreateRequest: RunCreateRequest, userId: UserId): Future[RunId]
 
   def getRunByIdAndUser(runId: RunId, userId: UserId): Future[Option[Run]]
 
   def deleteRunById(runId: RunId, userId: UserId): Future[Int]
 
-  def updateRun(request: RunUpdateRequest, userId: UserId): Future[Int]
+  def updateRun(runId: RunId, request: RunUpdateRequest, userId: UserId): Future[Int]
 
 }
 
 object RunService {
 
-  def apply(runRepository: RunRepository)(implicit executionContext: ExecutionContext): RunService =
+  def apply(runRepository: RunRepository, projectService: ProjectService)(
+    implicit executionContext: ExecutionContext
+  ): RunService =
     new RunService {
 
-      def addRun(runCreateRequest: RunCreateRequest): Future[RunId] = {
+      def addRun(runCreateRequest: RunCreateRequest, userId: UserId): Future[RunId] = {
 
         val newRun = Run(
           runId = RunId.random,
           projectId = runCreateRequest.projectId,
           projectVersion = runCreateRequest.projectVersion,
           status = Created,
           timeStart = Instant.now(),
-          userId = runCreateRequest.userId,
+          userId = userId,
           results = runCreateRequest.results
         )
-
-        runRepository.addRun(newRun)
+        projectService
+          .getUserProjectById(runCreateRequest.projectId, userId: UserId)
+          .flatMap(_ => runRepository.addRun(newRun))
```

Comment:
Although it is a reasonable change, let's make it as a separate task. Please create a ticket for it.

Reply:
Ok. I`ll rollback this changes and create new task.

Reply:
Done.